lib/render/SliceRenderer.cpp

```diff
@@ -0,0 +1,700 @@
+            else
+                dataValues[index+1] = 0.f;               
+            
+            dataValues[index]   = varValue;
+
+            index += 2;
+            coords[X] += deltas[X];
+        }
+        coords[Z] += deltas[Z];
+    }
+}
+    
+void SliceRenderer::_populateDataYZ(
+    float* dataValues,
+    Grid* grid
+) const {
+    std::vector<double> deltas = _calculateDeltas();
+    float varValue, missingValue; 
+    std::vector<double> coords(3, 0.0); 
+    coords[X] = _cacheParams.boxMin[X];
+    coords[Y] = _cacheParams.domainMin[Y];
+    coords[Z] = _cacheParams.domainMin[Z];
+
+    int index = 0;
+    for (int j=0; j<_textureHeight; j++) {
+        coords[Y] = _cacheParams.domainMin[Y];
+
+        for (int i=0; i<_textureWidth; i++) {
+            varValue = grid->GetValue(coords);
+            missingValue = grid->GetMissingValue();
+            if (varValue == missingValue) 
+                dataValues[index+1] = 1.f;
+            else
+                dataValues[index+1] = 0.f;               
+            
+            dataValues[index]   = varValue;
+
+            index += 2;
+            coords[Y] += deltas[Y];
+        }
+        coords[Z] += deltas[Z];
+    }
+}
+
+int SliceRenderer::_saveTextureData() {
+    Grid* grid = nullptr;
+    int rc = DataMgrUtils::GetGrids(
+        _dataMgr, 
+        _cacheParams.ts,
+        _cacheParams.varName,
+        _cacheParams.boxMin,
+        _cacheParams.boxMax,
+        true,
+        &_cacheParams.refinementLevel,
+        &_cacheParams.compressionLevel,
+        &grid
+    );
+
+    grid->SetInterpolationOrder(1);
+    
+    if (rc<0) {
+        SetErrMsg("Unable to acquire Grid for Slice texture");
+        return(rc);
+    }
+    assert(grid);
+
+    _setVertexPositions();
+
+    int textureSize = 2 * _textureWidth * _textureHeight;
+    float* dataValues    = new float[textureSize];
+
+    if      (_cacheParams.orientation == XY) 
+        _populateDataXY(dataValues, grid);
+    else if (_cacheParams.orientation == XZ) 
+        _populateDataXZ(dataValues, grid);
+    else if (_cacheParams.orientation == YZ) 
+        _populateDataYZ(dataValues, grid);
+    else    
+        assert(0);
+
+    _createDataTexture(dataValues);
+
+    delete [] dataValues;
+    delete grid;
+    grid = nullptr;
+
+    return rc;
+}
+
+void SliceRenderer::_createDataTexture(float* dataValues) {
+    if (_dataValueTextureID != 0) 
+        glDeleteTextures(1, &_dataValueTextureID);
+    
+    glGenTextures(1, &_dataValueTextureID);
+    glActiveTexture(GL_TEXTURE1);
+    glBindTexture(GL_TEXTURE_2D, _dataValueTextureID);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+
+    glTexImage2D(
+        GL_TEXTURE_2D, 
+        0, 
+        GL_RG32F, 
```

Comment:
A 64bit per pixel floating point texture is a little excessive here however this is not a big deal.